### ISSUE TITLE
ResponseChatBubble (de)serialization issue

### ISSUE THREAD

**Issue by arsenalol1** [api-ai/apiai-java-client#32]
Hi !

I encoutred a problem with serialization of the attribute ResponseChatBubble.textToSpeech.
Api.ai want this  : 
`{
          "type": "simple_response",
          "platform": "google",
          "textToSpeech": "This is a simple response."
        }`

But the class ResponseChatBubble generate this JSON : 
`{
          "items": [
            {
              "textToSpeech": "This is a simple response."
            }
          ],
          "type": "simple_response",
          "platform": "google"
        }`

Even the deserialization is not working because the ResponseChatBubble is not expecting an array but a string value. 

For testing : 

- Deserialization

`ResponseChatBubble response = (ResponseChatBubble) GsonFactory.getDefaultFactory().getGson().fromJson(
				"{ \"speech\": \"Hey!\",\"messages\": [{ \"type\": \"simple_response\", \"platform\": \"google\", \"textToSpeech\": \"This is a simple response.\" }]}",
				Fulfillment.class).getMessages().get(0);`

Here the object response does not contain the message "This is a simple response.".

- Serialization : 

`Item item = new Item();
 item.setTextToSpeech("This is a simple response.");
response.setItems(Lists.newArrayList(item));
String json = GsonFactory.getDefaultFactory().getGson().toJson(response);
`
The json value will be contain an array like a said before.

Thanks !

**Comment by folomeev** [comment]
Confirmed and fixed the deserialization issue. From now a JSON with no "items" array but having "textToSpeech" field would be deserialized correctly.

Speaking about serialization: 
```javascript
{ 
    "items": [ { "textToSpeech": "This is a simple response." } ], 
    "type": "simple_response", 
    "platform": "google"
}
```
is valid JSON and should be accepted by Api.ai without any problems.

**Comment by arsenalol1** [comment]
Thanks for the fix! 

For the serialization, that's not working ; here is what is sent to Api.ai (copy paste from 'JSON' button of Api.ai chat) : 
`    "fulfillment": {
      "speech": "",
      "messages": [
        {
          "items": [
            {
              "textToSpeech": "This is a simple response."
            }
          ],
          "type": "simple_response",
          "platform": "google"
        }
      ]
    }`

But nothing is shown : 
![image](https://user-images.githubusercontent.com/5845930/29018626-196f76ea-7b5c-11e7-9972-6431959c0fde.png)


Thanks for your help :)

**Comment by folomeev** [comment]
Okay, despite that such behavior is strange, the fix that change array of single item into object fields was added. 

It is available on a fresh release v1.6.12. Please update and check.

**Comment by arsenalol1** [comment]
The fix that change array of single item into object fields is working :) . Thanks ! 


I think that there is a wrong interpretation of some fields in API.AI. Some class like ResponseChatBubble or ResponseSpeech has got an array of String as attribute : I think that you choose an array because on API.AI webview, the input is also an array. What I've seen is that the array is to provide a random response. But, when I see the json response given by API.AI test chat, I've never seen an array but always the value that is randomly choosen. 

So, for me the conclusion is that API.AI is not waiting for an array, but for a unique value ; and in the web view, a string value is randomly choosen between value of the array **before** sending the response in JSON. So, for me, in java classes, we should not have array but string value. This is why we need to create some fixes on serialization and deserialisation stuff.

What do you think ? 

Thanks again :)
